lib/groupify/adapter/active_record/association_extensions.rb

```diff
@@ -0,0 +1,100 @@
+module Groupify
+  module ActiveRecord
+    module AssociationExtensions
+      extend ActiveSupport::Concern
+
+      # Defined to create alias methods before
+      # the association is extended with this module
+      def <<(*)
+        super
+      end
+
+      def add_without_exception(*children)
+        add_children_to_parent(children.flatten, false)
+      end
+
+      def add_with_exception(*children)
+        add_children_to_parent(children.flatten, true)
+      end
+
+      alias_method :add_as_usual, :<<
+      alias_method :<<, :add_without_exception
+      alias_method :add, :add_with_exception
+
+      def as(membership_type)
+        return self unless membership_type
+        merge(Groupify.group_membership_klass.as(membership_type))
+      end
+
+      def delete(*records)
+        remove_children_from_parent(records.flatten, :delete){ |*args| super(*args) }
```

Comment:
@dwbutler I don't really like how `super` is implemented here, but was having problems with getting a reference to `super` otherwise (e.g. `remove_children_from_parent(records.flatten, :delete, &method(:super))` or something like that didn't work... newer versions of Ruby allow `super_method`, but didn't want to go that route)